### DISCUSSION TITLE
Query result download link in dashboard Textbox redirects to an error page.

### DISCUSSION THREAD

**Discussion by cozy07** [getredash/redash#7016] [Q&A] [answered]
I have created a dashboard and added a Textbox to provide a link for downloading query results.
```
Markdown:
[Download Link](https://redash.example.com/api/queries/1/results.csv?api_key=xxxxxxxxxxxxxxxxxxxxx)
```

When I click on the created link, it redirects to a page displaying the following content:
`It seems like the page you're looking for cannot be found.`

Although it appears to be an error page, refreshing the screen initiates the download of the CSV file.

Is there a way to start the file download without redirecting to the error page?
It seems to be the same issue with it, but I could not find a workaround.
https://github.com/getredash/redash/issues/5046

Redash version: v10.1.0
Operating environment: ECS Fargate
Browser: The issue persists across all browsers tested

Thanks in advance.

**Comment by zachliu** [comment, marked as answer]
i can confirm it also happens in `24.06.0-dev`

* only "left click" triggers this issue
* "middle click" (open in a new page) downloads the data
* "right click" + "save as" downloads the data

this is because the generated link doesn't have a ` target="_blank"` attribute

the markdown-js lib has been out of maintenance for 5 years, see https://github.com/evilstreak/markdown-js/issues/186
too bad we can't simply add `{: target="_blank"}`

**Reply by cozy07** [reply]
Thank you for your response.
I understand that the only workaround is to right-click and save.For now, we will handle it in that way.
